compiler-rt/lib/sanitizer_common/sanitizer_common.h

```diff
@@ -290,6 +290,17 @@ bool SetEnv(const char *name, const char *value);
 
 u32 GetUid();
 void ReExec();
+
+#if defined(__ELF__) && !SANITIZER_FUCHSIA
+extern uptr kDynamic[] asm("_DYNAMIC");
+inline void DoesNotSupportStaticLinking() {
+  // This will fail to link with -static. However, -static-pie is not detected.
```

Comment:
Is there a TODO that should be added to detect -static-pie? How can that one be detected?

Reply:
I've clarified that we are unable to detect `-static-pie` at link time. 

This is not a TODO, since there is no way to statically detect `-static-pie`.